### ISSUE TITLE
az keyvault update does not work

### ISSUE THREAD

**Issue by WiseMack** [Azure/azure-cli#8172]
I was trying to turn "Set-AzureRmKeyVaultAccessPolicy" into "az keyvault update" so it matches with the rest of my code.

**Describe the bug**
The command az keyvault update -name $keyvaultName --enabled-for-deployment true
does not recognize the previously set variable $keyvaultName

Error message:
az: error: unrecognized arguments: airheadsbcKeys
usage: az [-h] [--verbose] [--debug] [--output {json,jsonc,table,tsv,yaml}]
          [--query JMESPATH]
          {keyvault} ...

**To Reproduce**
Create an empty keyvault and attempt to enable it for deployment

**Expected behavior**
Not an error

**Environment summary**
CLI version (`az --version 2.0`)

**Additional context**
When setting

$keyvaultName = "airheadsbcKeys"

This works

Set-AzureRmKeyVaultAccessPolicy -VaultName $keyvaultName -EnabledForDeployment

This doesn't work

az keyvault update -name $keyvaultName --enabled-for-deployment true

**Comment by WiseMack** [comment]
I believe I have found the problem here:
https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_params.py

In #L56 of the #region vault (management)
c.argument('vault_name', vault_name_type, options_list=['--name', '-n'])
vault_name is added to the keyvault
vault_base_url is NOT added to the keyvault

get_vault_base_url_type is part of the "from ._validators import " in #L20
vault_name_type is NOT part of the " from ._validators import " in #L20

For example in #L71 keyvault create
c.argument('vault_name', completer=None)
is used

Later in #L100 
c.argument('vault_base_url', vault_name_type, type=get_vault_base_url_type(self.cli_ctx), id_part=None)
keep in mind that vault_base_url is not part of the #L56 of the #region vault (management)

So in summary
when CREATING a keyvault we are using a "name" that is type "vault base url"
when USING a keyvault we are using a "vault base url" that doesn't have a type.

when searching for "vault base url" there are 9 hits.

**Comment by WiseMack** [comment]
https://github.com/Azure/azure-cli/pull/8175
I tried to fix the problem here.

**Comment by schaabs** [comment]
@WiseMack I think the issue here might be with your command and not argument parsing.  The az keyvault update command you have listed above has a - missing before the name argument.  Your command specifies -name when it should specify --name.  

I did some testing and when I omit one of the hyphens before name I get the same error as you did above. 

```
$ az keyvault create -n sstestupdatevault -l westus -g sswus2
{
  "id": "/subscriptions/6d8e19c2-3a60-481f-8ebb-118dc3007c5d/resourceGroups/sswus2/providers/Microsoft.KeyVault/vaults/sstestupdatevault",
  "location": "westus",
  "name": "sstestupdatevault",
  "properties": {
   ...
    "createMode": null,
    "enablePurgeProtection": null,
    "enableSoftDelete": null,
    "enabledForDeployment": false,
    "enabledForDiskEncryption": null,
    "enabledForTemplateDeployment": null,
    "networkAcls": null,
    "provisioningState": "Succeeded",
    "sku": {
      "name": "standard"
    },
    "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
    "vaultUri": "https://sstestupdatevault.vault.azure.net/"
  },
  "resourceGroup": "sswus2",
  "tags": {},
  "type": "Microsoft.KeyVault/vaults"
}
$ az keyvault update -name sstestupdatevault --enabled-for-deployment true
az: error: unrecognized arguments: sstestupdatevault
usage: az [-h] [--verbose] [--debug] [--output {json,jsonc,table,tsv,yaml}]
          [--query JMESPATH]
          {keyvault} ...
```

However when I fix the --name parameter the vault is updated correctly

```
$ az keyvault update --name sstestupdatevault --enabled-for-deployment true
{
  "id": "/subscriptions/6d8e19c2-3a60-481f-8ebb-118dc3007c5d/resourceGroups/sswus2/providers/Microsoft.KeyVault/vaults/sstestupdatevault",
  "location": "westus",
  "name": "sstestupdatevault",
  "properties": {
  ...
    "createMode": null,
    "enablePurgeProtection": null,
    "enableSoftDelete": null,
    "enabledForDeployment": true,
    "enabledForDiskEncryption": null,
    "enabledForTemplateDeployment": null,
    "networkAcls": null,
    "provisioningState": "Succeeded",
    "sku": {
      "name": "standard"
    },
    "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
    "vaultUri": "https://sstestupdatevault.vault.azure.net/"
  },
  "resourceGroup": "sswus2",
  "tags": {},
  "type": "Microsoft.KeyVault/vaults"
}
```

Can you check your command and see if the issue still repros when you specify --name rather than -name?

**Comment by tjprescott** [comment]
This is a usage error. `-name foo` would be interpreted by the CLI as `-n ame`, seeing "ame" as the name because the space between a short-option (with one -) and its value is optional. This is design argparse behavior upon which the CLI is based. thus `foo` would be an unrecognized argument, which is what your error message states.
